gusto/diagnostics.py

```diff
@@ -44,12 +50,8 @@ def max(f):
 """, "maxify"), f.dof_dset.set, fmax(op2.MAX), f.dat(op2.READ))
         return fmax.data[0]
 
-    @staticmethod
-    def rms(f):
-        V = FunctionSpace(f.function_space().mesh(), "DG", 1)
-        c = Function(V)
-        c.assign(1)
-        rms = sqrt(assemble(dot(f, f)*dx)/assemble(c*dx))
+    def rms(self, f):
+        rms = sqrt(assemble(dot(f, f)*dx)/self.area)
```

Comment:
Can you not just use `CellVolume`?

```python
def rms(f):
    return sqrt(assemble(dot(f, f)/CellVolume(f.ufl_domain()) * dx))
```

Reply:
Awesome - knew you'd have the answer! I did look at the `CellVolume` function but couldn't figure out what to give it. I'll close this and make a new PR to avoid all the junk commits...

Reply:
TypeError: unsupported operand type(s) for *: 'MeshGeometry' and 'Measure'

Reply:
ah, bracketing issue - sorted!

Reply:
OK, now I'm confused by the answer - see ipython output below:
```
In [9]: mesh=RectangleMesh(100, 100, 10, 1)

In [10]: V=FunctionSpace(mesh, "DG", 1)

In [11]: f=Function(V)

In [12]: c=Function(V).assign(1.)

In [13]: assemble(c*dx)
Out[13]: 10.000000000000732

In [14]: assemble(CellVolume(f.ufl_domain())*dx)
Out[14]: 0.0049999999999998362
```
The first seems to give the total area (which is what I want); the second seems to give the area of an element(?). What am I not understanding?!

Reply:
Every cell has area 0.5*10/10,000 = 5e-4.  The area of the mesh is 10.

You have integrated the "function" 5e-4 over a domain of area 10, giving you 5e-3.

Reply:
Looks like `CellVolume` is not what I want then.

Reply:
Oh wait, I was confused.  You're trying to compute the volume of the mesh.  Just do `assemble(1*dx(domain=f.ufl_domain())`